eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -106,10 +129,12 @@ case object TickPruneStaleChannels
   * Created by PM on 24/05/2016.
   */
 
-class Router(nodeParams: NodeParams, watcher: ActorRef, initialized: Option[Promise[Done]] = None) extends FSMDiagnosticActorLogging[State, Data] {
+class Router(val nodeParams: NodeParams, watcher: ActorRef, initialized: Option[Promise[Done]] = None) extends FSMDiagnosticActorLogging[State, Data] {
 
   import Router._
 
+  implicit val np = nodeParams
```

Comment:
This implicit feels a bit hackish...it's only used for `PrivateChannel` to implicitly get our `nodeId`, maybe changing `PrivateChannel` to `PrivateChannel(nodeId: PublicKey, remoteNodeId: PublicKey, etc)` would be cleaner?

Reply:
How about removing this line and replace the other line by:

```scala
PrivateChannel(remoteNodeId, None, None)(nodeParams)
```

Reply:
Actually went with your proposal.